src/sentry/static/sentry/app/views/organizationMonitors/monitorForm.jsx

```diff
@@ -0,0 +1,124 @@
+import SelectField from 'app/views/settings/components/forms/selectField';
+import TextCopyInput from 'app/views/settings/components/forms/textCopyInput';
+import TextField from 'app/views/settings/components/forms/textField';
+import {Panel, PanelBody, PanelHeader} from 'app/components/panels';
+import SentryTypes from 'app/sentryTypes';
+import {t, tct} from 'app/locale';
+
+import MonitorModel from './monitorModel';
+
+export default class MonitorForm extends Component {
+  static propTypes = {
+    monitor: SentryTypes.Monitor,
+    apiEndpoint: PropTypes.string.isRequired,
+    apiMethod: PropTypes.string.isRequired,
+    onSubmitSuccess: PropTypes.func.isRequired,
+  };
+
+
+  constructor(...args) {
+    super(...args);
+    const {monitor} = this.props;
+    this.form = new MonitorModel();
+    if (monitor) {
+        this.form.setInitialData({
+            name: monitor.name,
+            type: monitor.type,
+            ...this.formDataFromConfig(monitor.type, monitor.config)
+        });
+    }
+  }
+
+
+  formDataFromConfig(type, config) {
+    switch (type) {
+      case 'cron_job':
+        return {
+          'config.schedule_type': config.schedule_type,
+          'config.schedule': config.schedule,
+        }
+      default:
+        return {};
+    }
+  }
+
+  render() {
+      const {monitor} = this.props;
+    return (
+        <Access access={['project:write']}>
+          {({hasAccess}) => (
+              <Form
+                allowUndo
+                requireChanges
+                apiEndpoint={this.props.apiEndpoint}
+                apiMethod={this.props.apiMethod}
+                model={this.form}
+                initialData={this.form.initialData}
+                onSubmitSuccess={this.props.onSubmitSuccess}
+              >
+                <Panel>
+                  <PanelHeader>{t('Details')}</PanelHeader>
+
+                  <PanelBody>
+                    {monitor &&
+                    <Field label={t('ID')}>
+                      <div className="controls">
+                        <TextCopyInput>{monitor.id}</TextCopyInput>
+                      </div>
+                    </Field>
+                    }
+                    <TextField
+                      name="name"
+                      placeholder={t('My Cron Job')}
+                      label={t('Name')}
+                      disabled={!hasAccess}
+                      required
+                    />
+                  </PanelBody>
+                </Panel>
+              <Panel>
+                  <PanelHeader>{t('Config')}</PanelHeader>
+
+                  <PanelBody>
+                    <SelectField
+                      name="type"
+                      label={t('Type')}
+                      disabled={!hasAccess}
+                      choices={[['cron_job', 'Cron Job']]}
+                      required
+                    />
+                    {this.form.getValue('type') === 'cron_job' &&
```

Comment:
So this probably won't work unless you wrap this block with an `<Observer>` (from `mobx-react` package).

```
<Observer>
{() => {
return this.form.getValue('type') === 'cron_job' &&
                       <SelectField
                        name="config.schedule_type"
                        label={t('Schedule Type')}
                        disabled={!hasAccess}
                        choices={[['crontab', 'Crontab']]}
                        required
                      />
}}
</Observer>
```

(You could wrap the entire form, but that means the entire form re-renders when the `type` field changes, instead of just this field)

Reply:
What's Observer do exactly for the layman? Seems like thats what I need. Also are we intending to keep using mobx or should I do something thats future proof?

Reply:
Undetermined, I would say a non-mobx specific approach would be better (e.g. onChange and set react state).

Reply:
> What's Observer do exactly for the layman

Mobx `Observer` will watch for observable values to be accessed within the function, it keeps track of which ones get accessed and will trigger a re-render when the observable is changed. In this case, using `form.getValue()` looks up the observable fields value.

https://github.com/getsentry/sentry/blob/cc608632ee5f0921954502f3098f9a8c9d39b42c/src/sentry/static/sentry/app/views/settings/components/forms/model.jsx#L11-L14

Reply:
k im just gonna use Observer because i dont want to spend much time on this